scheduler/scheduler.go

```diff
@@ -176,23 +241,45 @@ func (s *scheduler) Start() error {
 	}
 	for _, session := range sessions {
 		s.scheduleSession(session)
+		if _, found := s.playgrounds[session.PlaygroundId]; !found {
+			playground, err := s.storage.PlaygroundGet(session.PlaygroundId)
+			if err != nil {
+				return err
+			}
+			s.playgrounds[playground.Id] = playground
+		}
 
 		instances, err := s.storage.InstanceFindBySessionId(session.Id)
 		if err != nil {
 			return err
 		}
 
 		for _, instance := range instances {
-			s.scheduleInstance(instance)
+			s.scheduleInstance(instance, session.PlaygroundId)
 		}
 	}
+
+	// Refresh playground conf every 5 minutes
+	s.schedulePlaygroundsUpdate()
+
 	s.event.On(event.SESSION_NEW, func(sessionId string, args ...interface{}) {
+		s.mx.Lock()
+		defer s.mx.Unlock()
+
 		log.Printf("EVENT: Session New %s\n", sessionId)
 		session, err := s.storage.SessionGet(sessionId)
 		if err != nil {
 			log.Printf("Session [%s] was not found in storage. Got %s\n", sessionId, err)
 			return
 		}
+		if _, found := s.playgrounds[session.PlaygroundId]; !found {
```

Comment:
Do we need to do this here also?. Isn't this be populated here: https://github.com/play-with-docker/play-with-docker/pull/218/files#diff-9667cfa1ad6b8b445794f7c2469f069fR244 before in Start function?

Reply:
We populate on Start for all sessions we retrieve from the DB. But then new sessions could be created, on different playgrounds, so we need to populate for the new playgrounds.

Reply:
Right.